profile/src/main/java/com/scottlogic/deg/profile/reader/CoreAtomicTypesConstraintReaderSource.java

```diff
@@ -302,7 +309,40 @@ else if (stringValidatedValue.isPresent()) {
                         ),
                         rules
                     )
+            ),
+            new ConstraintReaderMapEntry(
+                AtomicConstraintType.IS_FROM_FILE.getText(),
+                ".*",
+                (dto, fields, rules) -> {
+                    String value = ConstraintReaderHelpers.getValidatedValue(dto, String.class);
+
+                    InputStream streamFromPath = createStreamFromPath(value);
+                    Set<String> names = CsvInputStreamReader.retrieveLines(streamFromPath);
+                    closeStream(streamFromPath);
+
+                    Set<Object> downcastedNames = new HashSet<>(names);
+                    Field field = fields.getByName(dto.field);
+
+                    return new IsInSetConstraint(field, downcastedNames, rules);
+                }
             )
         );
     }
+
+    private static InputStream createStreamFromPath(String path) {
+        try {
+            return new FileInputStream(path);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
at this point, we should probably throw a validation exception that says to the user the file could not be found

Reply:
Is ValidationException preferable to an IOException in this case?